src/refactor/interactiveHandler.py

```diff
@@ -0,0 +1,112 @@
+            transactObj = schedulerTransaction(command, networkHandler)
+        elif tokens[self.serviceIdx] == "TIME_MANAGEMENT" and tokens[self.subserviceIdx] == "SET_TIME":
+            transactObj = setTimeTransaction(command, networkHandler)
+        else:
+            transactObj = baseTransaction(command, networkHandler)
+
+        return transactObj
+    
+class baseTransaction:
+    def __init__(self, command, networkHandler):
+        self.networkHandler = networkHandler
+        self.inputParse = inputParser()
+        self.returnParse = receiveParser()
+        self.command = command
+        self.pkt = self.inputParse.parseInput(self.command)
+        self.dst = self.pkt["dst"]
+        self.dport = self.pkt['dport']
+        self.args = self.pkt['args']
+
+    def send(self):
+        self.networkHandler.send(self.dst, self.dport, self.args)
+
+    def receive(self):
+        return self.networkHandler.receive(self.dst, self.dport, 10000)
+        
+    def parseReturnValue(self, data):
+        return self.returnParse.parseReturnValue(self.dport, data, len(data))
+
+    def execute(self):
+        self.send()
+        ret = self.receive()
+        return self.parseReturnValue(ret)
+
+class setTimeTransaction(baseTransaction):
+    def execute(self):
+        tokens = self.inputParse.lexer(self.command)
+        time_param = tokens[-2]
+        now = (int(time_param))
+        if (now == 0):
+            now = int(time.time())
+
+        tokens[-2] = str(now)
+        self.pkt = self.inputParse.parseInput("".join(tokens))
+        self.args = self.pkt['args']
+        self.send()
+        return self.parseReturnValue(self.receive())
+
+
+class schedulerTransaction(baseTransaction):
+    def execute(self):
+        tokens = self.inputParse.lexer(self.command)
+        file_param = tokens[-2]
+        f = open(file_param, "r")
+        cmdList = f.readlines()
+        packetEmbedder = embedPacket(cmdList, self.args)
+        self.args = packetEmbedder.embedCSP()
+        self.send()
+        return self.parseReturnValue(self.receive())
+
+class getHKTransaction(baseTransaction):
+    def execute(self):
+        print("WARN: I have no idea how HK receive works, it may not at all")
+        self.send()
+        rxDataList = []
+        while True:
+            ret = self.receive()
+            rxDataList.append(self.parseReturnValue(ret))
+            if ret[2] != 1:
+                break
+        return rxDataList
+
+class satcliTransaction(baseTransaction):
+    def execute(self):
+        response = ""
+        self.send()
+        while True:
+            ret = self.receive()
+            returnVal = self.parseReturnValue(ret)
+            response += "{}\n".format(returnVal['resp'].decode("ascii").rstrip())
+            if (returnVal['status']) == 0:
+                break
+        return response.strip()
+
+class dummyTransaction(baseTransaction):
+    def execute(self):
```

Comment:
I can implement this in another branch once I get to testing out the refactor on the website side